src/embezzler.ts

```diff
@@ -49,46 +51,131 @@ import { Macro, shouldRedigitize, withMacro } from "./combat";
+  #location?: Location;
+
+  constructor(macro: Macro, location?: Location) {
+    this.#macro = macro;
+    this.#location = location;
+  }
+
+  get macro(): Macro {
+    return this.#macro;
+  }
+
+  get location(): Location {
+    if (!this.#location) {
+      throw "Embezzler fight tried to access a location, but none was set";
+    } else {
+      return this.#location;
+    }
+  }
+}
 
 export class EmbezzlerFight {
+  name: string;
   available: () => boolean;
   potential: () => number;
-  run: (options: EmbezzlerFightOptions) => void;
+  execute: (options: EmbezzlerFightRunOptions) => void;
   requirements: Requirement[];
-  draggable: boolean;
-  name: string;
+  draggable?: DraggableFight;
+  canInitializeWandererCounters: boolean;
+  wrongEncounterName: boolean;
+  gregariousReplace: boolean;
+
   /**
    * This is the class that creates all the different ways to fight embezzlers
    * @classdesc Something goes here
    * @prop {string} name The name of the source of this fight, primarily used to identify special cases.
    * @prop {() => boolean} available Returns whether or not we can do this fight right now.
    * @prop {() => number} potential Returns the number of embezzlers we expect to be able to fight from this source.
    * @prop {(options: EmbezzlerFightOptions) => void} run This runs the combat, optionally using the provided location and macro. Location is used only by draggable fights.
-   * @prop {Requirement[]} requirements This is an array of requirements to do this fight
-   * @prop {boolean} [draggable=false] This tells the script if it is able the embezzler to a different zone
+   * @prop {EmbezzlerFightConfigOptions} options configuration options for this fight
    *
    */
   constructor(
     name: string,
     available: () => boolean,
     potential: () => number,
-    run: (options: EmbezzlerFightOptions) => void,
-    requirements: Requirement[] = [],
-    draggable = false
+    execute: (options: EmbezzlerFightRunOptions) => void,
+    options: EmbezzlerFightConfigOptions = {}
   ) {
     this.name = name;
     this.available = available;
     this.potential = potential;
-    this.run = run;
-    this.requirements = requirements;
-    this.draggable = draggable;
+    this.execute = execute;
+    this.requirements = options.requirements ?? [];
+    this.draggable = options.draggable;
+    this.canInitializeWandererCounters = options.canInitializeWandererCounters ?? true;
+    this.gregariousReplace = options.gregariousReplace ?? false;
+    this.wrongEncounterName = options.wrongEncounterName ?? this.gregariousReplace;
   }
+
+  run(options: { macro?: Macro; location?: Location } = {}): void {
+    const fightMacro = options.macro ?? embezzlerMacro();
+    if (this.draggable) {
+      this.execute(new EmbezzlerFightRunOptions(fightMacro, this.location(options.location)));
+    } else {
+      this.execute(new EmbezzlerFightRunOptions(fightMacro));
+    }
+  }
+
+  location(location?: Location): Location {
```

Comment:
I'm confused--doesn't this mean that `location()` for the final orb fight returns something other than `the dire warren`??

Reply:
No. This is only used for wanderers, as the jsdoc states.